src/prefect/environments/execution/dask/k8s.py

```diff
@@ -162,25 +164,17 @@ def setup(self, storage: "Docker") -> None:  # type: ignore
                 )
 
     def execute(  # type: ignore
-        self, storage: "Docker", flow_location: str, **kwargs: Any
+        self, flow: "Flow", **kwargs: Any
     ) -> None:
         """
         Create a single Kubernetes job that spins up a dask scheduler, dynamically
         creates worker pods, and runs the flow.
 
         Args:
-            - storage (Docker): the Docker storage object that contains information relating
-                to the image which houses the flow
-            - flow_location (str): the location of the Flow to execute
+            - flow (Flow): the Flow object
             - **kwargs (Any): additional keyword arguments to pass to the runner
-
-        Raises:
-            - TypeError: if the storage is not `Docker`
         """
-        if not isinstance(storage, Docker):
-            raise TypeError("CloudEnvironment requires a Docker storage option")
-
-        self.create_flow_run_job(docker_name=storage.name, flow_file_path=flow_location)
+        self.create_flow_run_job(docker_name=get_flow_image(flow))
```

Comment:
It seems like having a dedicated `create_flow_run_job` is not really necessary anymore if all `execute` does is immediately call this?  Thoughts on removing that method?

Reply:
Yeah it could be moved into this function thanks for the callout